### PR TITLE
feat(protocol): add `DevnetTaikoL1`

### PR DESCRIPTION
**Pull request by davidtaikocha**

### CODE REVIEW

File: packages/protocol/contracts/devnet/DevnetTaikoL1.sol [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            blockRingBufferSize: 360_000,
+            maxBlocksToVerify: 16,
+            blockMaxGasLimit: 240_000_000,
+            livenessBond: 125e18, // 125 Taiko token
+            stateRootSyncInternal: 16,
+            maxAnchorHeightOffset: 64,
+            basefeeAdjustmentQuotient: 8,
+            basefeeSharingPctg: 75,
+            gasIssuancePerSecond: 5_000_000,
+            ontakeForkHeight: 0
```
Comment (YoGhurt111) [inline packages/protocol/contracts/devnet/DevnetTaikoL1.sol:24]: Let‘s try 20?

Comment (davidtaikocha) [reply packages/protocol/contracts/devnet/DevnetTaikoL1.sol:24]: Lets do `0` by default for client testing purpose? or `1`?

Comment (davidtaikocha) [reply packages/protocol/contracts/devnet/DevnetTaikoL1.sol]: Changed to `2`

Comment (YoGhurt111) [reply packages/protocol/contracts/devnet/DevnetTaikoL1.sol]: In my opinion, we need to make sure that the logic before and after ontake is correct every time we reset devnet

**Comment by YoGhurt111** [comment]
Need to `forge fmt` again

**Comment by davidtaikocha** [comment]
> Need to `forge fmt` again

Seems its auto-formatted by the ci?

**Comment by YoGhurt111** [comment]
> > Need to `forge fmt` again
> 
> Seems its auto-formatted by the ci? [9e5ae67](https://github.com/taikoxyz/taiko-mono/commit/9e5ae67d17b8ade28f6c506d01edad4a5a68681a)

Seems so....

**Review by YoGhurt111 — APPROVED** [review]

**Review by dantaik — APPROVED** [review]

**Pull request merged**